src/ota.rs

```diff
@@ -220,16 +217,29 @@ impl EspOta {
     pub fn initiate_update(&mut self) -> Result<&mut EspOtaUpdate, EspError> {
         self.check_read()?;
 
-        let partition = unsafe { esp_ota_get_next_update_partition(ptr::null()) };
+        // TODO: The documentation of `esp_ota_get_next_update_partition` states, that the returned
+        // pointer may be a null pointer, which indicates that:
+        // "indicates invalid OTA data partition, or that no eligible OTA app slot partition was found."
+        //
+        // This should probably be handled in here and throw a proper EspError.
+        let update_partition = unsafe { esp_ota_get_next_update_partition(ptr::null()) };
```

Comment:
I wasn't sure what kind of error should be thrown in case of a null ptr, hence the `TODO` block.

Reply:
Just pick something appropriate from the `ESP_ERR_*` error constants, and return it wrapped as an `EspError` instance. This way we will ensure, that `initiate_update` will complete successfully only if there was an update partition found.

Reply:
`ESP_ERR_NOT_SUPPORTED` and `ESP_ERR_INVALID_STATE` are both good candidates to return as an error code if you get a null pointer from ESP IDF. Perhaps just pick one of these two and return it.

Reply:
As it turns out, this isn't even necessary, as a null pointer will trigger an error when provided to [esp_ota_begin](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/ota.html#_CPPv413esp_ota_beginPK15esp_partition_t6size_tP16esp_ota_handle_t).
The current logic is sound and I added an inline comment to prevent further confusion